### PR TITLE
(fixes #7, fixes #9) support retry on Either and expoential backoff p…

### LINKED ISSUE #7: Support Either

### LINKED ISSUE #9: Add exponential back off policy (no jitter)

### PR DESCRIPTION
**Pull request by realstraw**
…olicy

### CODE REVIEW

File: src/main/scala/com/krux/stubborn/policy/ExponentialBackoff.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.krux.stubborn.policy
+
+
+trait ExponentialBackoff extends Policy {
+
+  def base: Int = ExponentialBackoffAndJitter.defaultBase
```
Comment (sethyates) [inline src/main/scala/com/krux/stubborn/policy/ExponentialBackoff.scala]: This references a child class's object properties, probably not the best.

File: src/main/scala/com/krux/stubborn/policy/ExponentialBackoffAndJitter.scala [hunk 2]
```diff
@@ -7,18 +7,13 @@ import scala.util.Random
  * Implements the FullJitter algorithm in
  * https://www.awsarchitectureblog.com/2015/03/backoff.html
  */
-trait ExponentialBackoffAndJitter extends Policy {
+trait ExponentialBackoffAndJitter extends ExponentialBackoff with Policy {
 
-  def base: Int = ExponentialBackoffAndJitter.defaultBase
+  override def base: Int = ExponentialBackoffAndJitter.defaultBase
```
Comment (sethyates) [inline src/main/scala/com/krux/stubborn/policy/ExponentialBackoffAndJitter.scala]: Given this is exactly the same as `ExponentialBackoff`, why override these?

File: src/main/scala/com/krux/stubborn/policy/ExponentialBackoff.scala (hunk 1 above)
Comment (realstraw) [inline src/main/scala/com/krux/stubborn/policy/ExponentialBackoff.scala]: oops, copy paste error, will update

**Comment by realstraw** [comment]
@sethyates addressed your comments

**Comment by sethyates** [comment]
Comments would be nice but other than that LGTM

**Comment by realstraw** [comment]
I will add more comments and docs next time

**Pull request merged**